proposals/lip-0048.md

```diff
@@ -25,15 +25,57 @@ This LIP defines the fee system in a modular way, as currently [used in the Lisk
+#### Fee Token ID
+
+Each chain can configure the token used to pay fees, as defined by `TOKEN_ID_FEE`. On the Lisk mainchain, the token used for transaction fees is the LSK token.
+
+#### Minimum Fee per Transaction
+
+As introduced in [LIP 0013][lip-0013], all transactions must have a transaction fee that is greater than or equal to a *minimum fee* that is always deducted from the part of the fee that goes to the block generator. The minimum fee, unless it is zero, is intended to discourage validators from sending zero-cost transactions in the blocks they generate.
+
+For a transaction `trs`, the minimum fee is determined by its size and the proportionality constant `MIN_FEE_PER_BYTE`: 
+```python
+trsSize = length(encodeTransaction(trs))
+minFee = MIN_FEE_PER_BYTE * trsSize
+```
+Note that chains can freely configure their `MIN_FEE_PER_BYTE`, which can even be set to zero. The `MIN_FEE_PER_BYTE` value cannot be changed, unless a hard fork occurs. For this reason, we allow sidechains to define a period of `MAX_BLOCK_HEIGHT_ZERO_FEE_PER_BYTE` blocks after the genesis block, for which the minimum fee does not apply. 
+
+This could, for example, be useful for sidechains that want to use LSK as the fee token. Initially, there is no LSK on the sidechain, so the sidechain cannot process transactions unless `MIN_FEE_PER_BYTE` is set to zero, which is generally not desirable since it essentially disables the feature of minimum fee for the chain. By granting an initial exemption period of `MAX_BLOCK_HEIGHT_ZERO_FEE_PER_BYTE` blocks, users have a window of time to transfer LSK tokens to the sidechain, after which the initially set value of `MIN_FEE_PER_BYTE` would start being applied. 
+
+### Method `payFee`
+
+In principle, any transaction that results in a state increase or significant computational effort could require a fee. As mentioned above, since current implementation lacks an automated way of charging fees in these cases, we introduce the [`payFee`](#payfee) method. 
+
+This method can be invoked by other modules to charge an additional fee when a state increase is triggered during the execution of a transaction, e.g., when a new validator is registered in the PoS module, a new sidechain is registered, or a user account is initialized in the Token module. Also, developers may decide to call the `payFee` method in transactions requiring an extensive computation, e.g., in verifying an inclusion proof.
+
+For example, this method is called during the execution of the [validator registration command][lip-0057#execution] in LIP 0057: 
+```python
+Fee.payFee(VALIDATOR_REGISTRATION_FEE)
+```
+
+### Fee Processing
+
+A part of the fee equal to the sum of the minimum fee and, if applicable, the initialization fee, is referred to as the *consumed fee*. [After the transaction is executed](#after-command-execution), consumed fee is either burned or transferred to a dedicated pool address. 
```

Comment:
I think here we should be more precise. This 'consumed fee' would be minimum fee plus other fees paid using the `payFee` method, which could be more general than initialization fees.

Reply:
```suggestion
We call the part of the fee used during the execution of a transaction (i.e., the sum of minFee plus the fees handled by payFee) as the "consumed" part of the fee.
```

Reply:
Namely, I would not discuss the burn vs pool here since it destructs the flow. 

1) Define consumed, available
2) Say that available goes to validator
3) Discuss the burn vs pool.

Reply:
Agreed and changed as suggested.